libgin/archive/tar.go

```diff
@@ -0,0 +1,133 @@
+package archive
+
+import (
+	"archive/tar"
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/G-Node/libgin/libgin/annex"
+	"github.com/gogs/git-module"
+)
+
+type TarWriter struct {
+	Repository *git.Repository
+	Commit     *git.Commit
+	writer     *tar.Writer
+}
+
+func NewTarWriter(repo *git.Repository, commit *git.Commit) *TarWriter {
+	return &TarWriter{Repository: repo, Commit: commit}
+}
+
+func (a *TarWriter) Write(target string) error {
+	tree := &a.Commit.Tree
+
+	gzipfile, err := os.Create(target)
+	if err != nil {
+		return err
+	}
+	defer gzipfile.Close()
+	gzipWriter := gzip.NewWriter(gzipfile)
+	defer gzipWriter.Close()
+
+	a.writer = tar.NewWriter(gzipWriter)
+	defer a.writer.Close()
+
+	return a.addTree(tree, "")
+}
+
+func (a *TarWriter) addBlob(blob *git.Blob, fname string) error {
+	header := tar.Header{
+		Name:    fname,
+		ModTime: time.Now(), // TODO: use commit time
+	}
+	var filemode os.FileMode
+	filemode |= 0660
+	if blob.IsLink() {
+		filemode |= os.ModeSymlink
+	}
+	blob.Blob()
+
+	stdout := new(bytes.Buffer)
+	stderr := new(bytes.Buffer)
+	blob.DataPipeline(stdout, stderr)
+	size := blob.Size()
+	reader := bufio.NewReader(stdout)
+	readbuf := make([]byte, 10240)
+	if annex.IsAnnexFile(blob) {
+		n, err := reader.Read(readbuf)
```

Comment:
Found an ignored `err`. Same in the `zip.addBlob` method methinks.

Reply:
Good catch.  Thanks.